components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java

```diff
@@ -0,0 +1,498 @@
+                        + MultitenantUtils.getTenantDomain(userName);
+                log.debug(msg, e);
+            }
+            throw handleFederatedAssociationManagerServerException(ERROR_WHILE_DELETING_FEDERATED_ASSOCIATION_OF_USER
+                    , e,
+                    true);
+        }
+    }
+
+    private void validateExistenceOfFederatedAssociations(String userName) throws FederatedAssociationManagerException {
+
+        if (!isValidFederatedAssociationsExist(userName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Valid federated associations does not exist for the user: " + userName);
+            }
+            throw handleFederatedAssociationManagerClientException(FEDERATED_ASSOCIATION_DOES_NOT_EXISTS, null, true);
+        }
+    }
+
+    private boolean isValidFederatedAssociationsExist(String userName) throws FederatedAssociationManagerException {
+
+        FederatedAssociation[] federatedUserAccountAssociationDTOS
+                = getFederatedAssociationsOfUser(userName);
+        return !ArrayUtils.isEmpty(federatedUserAccountAssociationDTOS);
+    }
+
+    private void validateFederatedAssociation(String userName, String federatedAssociationId)
+            throws FederatedAssociationManagerException {
+
+        if (StringUtils.isEmpty(federatedAssociationId)
+                || !isValidFederatedAssociation(userName, federatedAssociationId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("A valid federated association does not exist for the Id: " + federatedAssociationId
+                        + ", of the user: " + userName);
+            }
+            throw handleFederatedAssociationManagerClientException(INVALID_FEDERATED_ASSOCIATION, null, true);
+        }
+    }
+
+    private boolean isValidFederatedAssociation(String userName, String federatedAssociationId)
+            throws FederatedAssociationManagerException {
+
+        FederatedAssociation[] federatedUserAccountAssociationDTOS
+                = getFederatedAssociationsOfUser(userName);
+        if (federatedUserAccountAssociationDTOS != null) {
+            for (FederatedAssociation federatedUserAccountAssociationDTO
+                    : federatedUserAccountAssociationDTOS) {
+                if (federatedAssociationId.equals(federatedUserAccountAssociationDTO.getId())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private String getPlainUserName(String userName) {
+
+        String userNameWithoutDomain = getUsernameWithoutDomain(userName);
+        return removeTenantDomain(userNameWithoutDomain);
+    }
+
+    private String getUsernameWithoutDomain(String username) {
+
+        int index = username.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
+        if (index < 0) {
+            return username;
+        }
+        return username.substring(index + 1);
+    }
+
+    private String removeTenantDomain(String username) {
```

Comment:
MultitenantUtils.getTenantAwareUsername() will do the same thing

Reply:
Fixed in 78f08459d2c398277e992c0c738443cfe4edad1b